### PR TITLE
Main menu

### PR DESCRIPTION
**Pull request by matiasschwalb**
## Summary ##

Main menu, UITableView for displaying my custom cells containing book information.

### iPhone 5s

Simulator Screen Shot - iPhone 5s - 2019-05-24 at 15 43 50

### iPhone X/XS/XR

Simulator Screen Shot - iPhone XR - 2019-05-24 at 15 39 40

## Screencast ##

- none

## Known issues ##

Cell's height not adaptive (hardcoded). Can't figure out how to make cell heights dynamic in function to it's contents

### CODE REVIEW

File: WBooks/LibraryView.swift [hunk 1]
```diff
@@ -12,6 +12,7 @@ import WolmoCore
 
 class LibraryView: UIView, NibLoadable {
     
+    @IBOutlet weak var tblBooks: UITableView!
```
Comment (gmazzei) [inline WBooks/LibraryView.swift]: Change `tblBooks` to `tableBooks`. It's always better to write complete names.

File: WBooks/LibraryView.swift [hunk 2]
```diff
@@ -22,7 +23,9 @@ class LibraryView: UIView, NibLoadable {
 private extension LibraryView {
     
     func setupView() {
-        // TODO
+        let tableBackgroundColor = UIColor(displayP3Red: 231.0 / 255.0, green: 245.0 / 255.0, blue: 249.0 / 255.0, alpha: 1)
+        self.layer.backgroundColor = tableBackgroundColor.cgColor
```
Comment (gmazzei) [inline WBooks/LibraryView.swift]: Use `self.` only when needed. We can remove it here.

File: WBooks/LibraryViewController.swift [hunk 3]
```diff
@@ -9,34 +9,75 @@
 import UIKit
 
 final class LibraryViewController: UIViewController {
-    
+
     private let libraryView: LibraryView = LibraryView.loadFromNib()!
     
+    let spacingBetweenCells: CGFloat = 10
```
Comment (gmazzei) [inline WBooks/LibraryViewController.swift]: We can make this a `static` constant.

File: WBooks/LibraryViewController.swift [hunk 4]
```diff
@@ -9,34 +9,75 @@
...
-        view.backgroundColor = UIColor.white
-        
-        let frame = CGRect(x: 0, y: view.frame.size.height / 2 - 11, width: view.frame.size.width, height: 22)
-        let label = UILabel(frame: frame)
-        label.text = "Welcome! This is a new blank project"
-        label.textAlignment = .center
-        label.sizeToFit()
-        view.addSubview(label)
-        */
+        let nib = UINib.init(nibName: "MyCustomCell", bundle: nil)
```
Comment (gmazzei) [inline WBooks/LibraryViewController.swift]: I think that the `init` part can be removed. We can write it like this: `UINib(nibName: "MyCustomCell", bundle: nil)`

File: WBooks/LibraryViewController.swift [hunk 5]
```diff
@@ -9,34 +9,75 @@
...
+    // Make the background color show through
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let headerView = UIView()
+        headerView.backgroundColor = UIColor.clear
+        return headerView
     }
+    // Cell generator
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "MyCustomCell", for: indexPath) as? MyCustomCell else {
```
Comment (gmazzei) [inline WBooks/LibraryViewController.swift]: Change "MyCustomCell" to a static constant. Also, try to find a more suitable name for `MyCustomCell` (something like `LibraryCell` perhaps?)

File: WBooks/MyCustomCell.swift [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        
+        topLabel.font = UIFont.boldSystemFont(ofSize: 20)
+        botLabel.font = UIFont.systemFont(ofSize: 12)
+        
+        topLabel.lineBreakMode = .byWordWrapping
+        topLabel.numberOfLines = 0
+    
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
```
Comment (gmazzei) [inline WBooks/MyCustomCell.swift]: If this method does nothing, we should delete it.

File: WBooks/LibraryView.swift (hunk 1 above)
Comment (matiasschwalb) [reply WBooks/LibraryView.swift]: Done

File: WBooks/LibraryView.swift (hunk 2 above)
Comment (matiasschwalb) [reply WBooks/LibraryView.swift]: Done

File: WBooks/MyCustomCell.swift (hunk 6 above)
Comment (matiasschwalb) [reply WBooks/MyCustomCell.swift]: I'll leave it commented. Will use later

**Pull request merged**